ethers-etherscan/src/account.rs

```diff
@@ -0,0 +1,532 @@
+#[serde(rename_all = "camelCase")]
+pub struct ERC721TokenTransferEvent {
+    pub block_number: String,
+    pub time_stamp: String,
+    pub hash: String,
+    pub nonce: String,
+    pub block_hash: String,
+    pub from: String,
+    pub contract_address: String,
+    pub to: String,
+    #[serde(rename = "tokenID")]
+    pub token_id: String,
+    pub token_name: String,
+    pub token_symbol: String,
+    pub token_decimal: String,
+    pub transaction_index: String,
+    pub gas: String,
+    pub gas_price: String,
+    pub gas_used: String,
+    pub cumulative_gas_used: String,
+    pub input: String,
+    pub confirmations: String,
+}
+
+/// The raw response from the mined blocks API endpoint
+#[derive(Debug, Serialize, Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct MinedBlock {
+    pub block_number: String,
+    pub time_stamp: String,
+    pub block_reward: String,
+}
+
+/// The pre-defined block parameter for balance API endpoints
+pub enum Tag {
+    Earliest,
+    Pending,
+    Latest,
+}
+
+impl ToString for Tag {
+    fn to_string(&self) -> String {
+        match self {
+            Tag::Earliest => "earliest",
+            Tag::Pending => "pending",
+            Tag::Latest => "latest",
+        }
+        .to_string()
+    }
+}
+
+impl Default for Tag {
+    fn default() -> Self {
+        Tag::Latest
+    }
+}
+
+/// The list sorting preference
+pub enum Sort {
+    Asc,
+    Desc,
+}
+
+impl ToString for Sort {
+    fn to_string(&self) -> String {
+        match self {
+            Sort::Asc => "asc",
+            Sort::Desc => "desc",
+        }
+        .to_string()
+    }
+}
+
+/// Common optional arguments for the transaction or event list API endpoints
+pub struct TxListParams {
+    start_block: u64,
+    end_block: u64,
+    page: u64,
+    offset: u64,
+    sort: Sort,
+}
+
+impl TxListParams {
+    pub fn new(start_block: u64, end_block: u64, page: u64, offset: u64, sort: Sort) -> Self {
+        Self { start_block, end_block, page, offset, sort }
+    }
+}
+
+impl Default for TxListParams {
+    fn default() -> Self {
+        Self { start_block: 0, end_block: 99999999, page: 0, offset: 10000, sort: Sort::Asc }
+    }
+}
+
+impl From<TxListParams> for HashMap<&'static str, String> {
+    fn from(tx_params: TxListParams) -> Self {
+        let mut params = HashMap::new();
+        params.insert("startBlock", tx_params.start_block.to_string());
+        params.insert("endBlock", tx_params.end_block.to_string());
+        params.insert("page", tx_params.page.to_string());
+        params.insert("offset", tx_params.offset.to_string());
+        params.insert("sort", tx_params.sort.to_string());
+        params
+    }
+}
+
+/// Options for querying internal transactions
+pub enum InternalTxQueryOption<A: AsRef<str>> {
+    ByAddress(A),
+    ByTransactionHash(A),
+    ByBlockRange,
+}
+
+/// Options for querying ERC20 or ERC721 token transfers
+pub enum TokenQueryOption<A: AsRef<str>> {
```

Comment:
I think you can omit the `: AsRef<str>` bound here, since the type is restricted in the API calls as well